src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -653,6 +702,30 @@ private void PopulateProtocol()
             }
         }
 
+        public static string GetLocalDBInstance(string dataSource, out bool error)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(dataSource), "Empty data source");
+
+            error = false;
+
+            string instanceName = null;
+
+            string workingDataSource = dataSource.ToLowerInvariant();
+
+            string[] tokensByBackSlash = workingDataSource.Split(BackSlashSeparator);
+
+            // All LocalDb endpoints are of the format host\instancename where host is always (LocalDb) (case-insensitive)
+            if (tokensByBackSlash.Length == 2 && LocalDbHost.Equals(tokensByBackSlash[0].Trim()))
+            {
+                if (!string.IsNullOrEmpty(tokensByBackSlash[1].Trim()))
```

Comment:
Can call `String.IsNullOrWhiteSpace` to avoid allocations when calling `Trim`. https://msdn.microsoft.com/en-us/library/system.string.isnullorwhitespace(v=vs.110).aspx

According to MSDN, both `IsNullOrWhiteSpace` and `Trim` use the same character set for detecting white space, so the check would be identical.